usr/src/cmd/zoneadm/svc-zones

```diff
@@ -232,37 +220,67 @@ case "$1" in
 	SVC_TIMEOUT=`svcprop -p stop/timeout_seconds $SMF_FMRI`
 
 	#
-	# First, try shutting down any running zones for which an "init 0" may
-	# work.
+	# First, try shutting down any running zones, using the best-known
+	# clean-shutdown methods. This varies by zone brand.
 	#
 	MAXSHUT=`expr 3 \* $SVC_TIMEOUT \/ 4` # 3/4 of time to zone shutdown
 	MAXHALT=`expr $SVC_TIMEOUT \/ 4`      # rest of time goes to halt
 
-	zonelist=`shutdown_zones`
+	zonelist=`zoneadm list -p | grep -v ^0:global`
 
 	if [ -n "$zonelist" ]; then
 		SHUTDOWN=0
 		echo "Shutting down running zones (for up to $MAXSHUT" \
 		    "seconds):\c"
 
-		for zone in $zonelist; do
-			echo " $zone\c"
-			zlogin -S $zone /sbin/init 0 < /dev/null >&0 2>&0 &
+		# Try and do any special brand-specific shutdown procedures,
+		# otherwise treat the zone like a native zone where you can
+		# both `zlogin -S` and `/sbin/init 0` in-zone.
+		#
+		# This is new with OS-8684, and new brands or changes in
+		# brand implementations will require revisiting this.
+		for zonedata in $zonelist; do
+			zone=`echo $zonedata | awk -F: '{print $2}'`
+			brand=`echo $zonedata | awk -F: '{print $6}'`
+			echo " $zone($brand)\c"
+			if [ "$brand" == "lx" ]; then
```

Comment:
Also, NGL, in a PERFECT world we'd be able to invoke something like `/usr/lib/${brand}/...`, or even just make sure `zoneadm -z $zone shutdown...` works for every brand, but in SmartOS that may involve SO MANY gymnastics it's probably best to keep it straightforward, if a little copy-and-paste-y.